compiler/rustc_lint/src/internal.rs

```diff
@@ -263,6 +264,49 @@ fn gen_args(segment: &PathSegment<'_>) -> String {
     String::new()
 }
 
+declare_tool_lint! {
+    /// The `non_glob_import_of_type_ir_inherent_item` lint detects
+    /// non-glob imports of module `rustc_type_ir::inherent`.
+    pub rustc::NON_GLOB_IMPORT_OF_TYPE_IR_INHERENT,
+    Allow,
+    "non-glob import of `rustc_type_ir::inherent`",
+    report_in_external_macro: true
+}
+
+declare_lint_pass!(TypeIr => [NON_GLOB_IMPORT_OF_TYPE_IR_INHERENT]);
```

Comment:
I'm inclined to consolidate the lint passes `TypeIr` and `TyTyKind` to reduce bloat. Just need to think of a good name :thinking:

Reply:
"opinionated imports"?